pkg/ingress/controller/apisix_tls.go

```diff
@@ -0,0 +1,186 @@
+		kubeclientset:   kubeclientset,
+		apisixClientset: apisixTlsClientset,
+		apisixTlsList:   apisixTlsInformer.Lister(),
+		apisixTlsSynced: apisixTlsInformer.Informer().HasSynced,
+		workqueue:       workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ApisixTlses"),
+	}
+	apisixTlsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    controller.addFunc,
+			UpdateFunc: controller.updateFunc,
+			DeleteFunc: controller.deleteFunc,
+		})
+	return controller
+}
+
+func (c *ApisixTlsController) Run(stop <-chan struct{}) error {
+	if ok := cache.WaitForCacheSync(stop); !ok {
+		log.Errorf("sync ApisixService cache failed")
+		return fmt.Errorf("failed to wait for caches to sync")
+	}
+	go wait.Until(c.runWorker, time.Second, stop)
+	return nil
+}
+
+func (c *ApisixTlsController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *ApisixTlsController) processNextWorkItem() bool {
+	defer recoverException()
+	obj, shutdown := c.workqueue.Get()
+	if shutdown {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer c.workqueue.Done(obj)
+		var key string
+		var ok bool
+
+		var tqo *TlsQueueObj
+		if tqo, ok = obj.(*TlsQueueObj); !ok {
+			c.workqueue.Forget(obj)
+			return fmt.Errorf("expected TlsQueueObj in workqueue but got %#v", obj)
+		}
+		if err := c.syncHandler(tqo); err != nil {
+			return fmt.Errorf("error syncing '%s': %s", key, err.Error())
+		}
+
+		c.workqueue.Forget(obj)
+		return nil
+	}(obj)
+	if err != nil {
+		runtime.HandleError(err)
+	}
+	return true
+}
+
+func (c *ApisixTlsController) syncHandler(tqo *TlsQueueObj) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(tqo.Key)
+	if err != nil {
+		log.Errorf("invalid resource key: %s", tqo.Key)
+		return fmt.Errorf("invalid resource key: %s", tqo.Key)
+	}
+	apisixTlsYaml := tqo.OldObj
+	if tqo.Ope != state.Delete {
+		apisixTlsYaml, err = c.apisixTlsList.ApisixTlses(namespace).Get(name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.Infof("apisixTls %s is removed", tqo.Key)
+				return nil
+			}
+			runtime.HandleError(fmt.Errorf("failed to list apisixTls %s/%s", tqo.Key, err.Error()))
+			return err
+		}
+	}
+	apisixTls := apisix.ApisixTlsCRD(*apisixTlsYaml)
+	sc := &apisix.SecretClient{}
+	if tls, err := apisixTls.Convert(sc); err != nil {
+		return err
+	} else {
+		// sync to apisix
+		log.Debug(tls)
+		log.Debug(tqo)
+		state.SyncSsl(tls, tqo.Ope)
+	}
+	return err
+}
+
+func (c *ApisixTlsController) addFunc(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	rqo := &TlsQueueObj{Key: key, OldObj: nil, Ope: state.Create}
+	c.workqueue.AddRateLimited(rqo)
+}
+
+func (c *ApisixTlsController) updateFunc(oldObj, newObj interface{}) {
+	oldTls := oldObj.(*apisixV1.ApisixTls)
+	newTls := newObj.(*apisixV1.ApisixTls)
+	if oldTls.ResourceVersion == newTls.ResourceVersion {
```

Comment:
This logic is changed in https://github.com/apache/apisix-ingress-controller/pull/103

Reply:
after merge this PR, will do in the #103

Reply:
OK